### DISCUSSION TITLE
Why my built fail when i try to deploy it to vercel and when I use canvas? (Nodejs)

### DISCUSSION THREAD

**Discussion by xts-bit** [vercel/community#2151] [Help] [answered]
### Summary

Why my built fail when I try to deploy it to Vercel and when I use Canvas? (Nodejs) I am trying to generate a blurhash string of an image in my Nodejs backend so to do that I need to install [blurhash](https://www.npmjs.com/package/blurhash) and [canvas](https://www.npmjs.com/package/canvas) to do that so when I write a function that does that and it works well in my local machine without any error 

So after that, I deployed it to Vercel, and then when I go to my website URL it gave me "This Serverless Function has crashed.
Your connection is working correctly. Vercel is working correctly." so I quickly open my Vercel dashboard and when open my Vercel logs and I saw this error coming 

Error: 
Error: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /var/task/node_modules/canvas/build/Release/libglib-2.0.so.0)
    at Module._extensions..node (node:internal/modules/cjs/loader:1302:18)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/var/task/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12) {
  code: 'ERR_DLOPEN_FAILED'
}
RequestId: 56dc8fb9-7cdf-493e-89cd-df30b88b51df Error: Runtime exited with error: exit status 1
Runtime.ExitError

How can I fix this problem and make things work?

### Additional information

After searching about this issue I found that "This error message indicates that the libz.so.1 library with version ZLIB_1.2.9 is missing, which is required by the libpng16.so.16 library in the canvas module." 

My Function:
```
async function generateBlurhashString(post) {
    const image = await loadImage(post);
    const canvas = createCanvas(image.width, image.height);
    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0);
    const imageData = context.getImageData(0, 0, image.width, image.height);
    return blurhash.encode(imageData.data, imageData.width, imageData.height, 4, 4);
}
```

**Comment by swarnava** [comment]
Thanks for reaching out to Vercel Support! 

It looks like you're running in to an error because your project requires a package that is not already available in the [Vercel build image](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#build-image). You can install the necessary package using [yum install](https://vercel.com/docs/concepts/deployments/troubleshoot-a-build#installing-something-else-in-the-build-image) by overriding the [Install command](https://vercel.com/docs/concepts/deployments/configure-a-build#install-command) in the project settings on Vercel.

You'll need to override this with `yum install [package-name] -y && npm install`

**Reply by xts-bit** [reply]
@swarnava So do i need to use docker?

**Reply by swarnava** [reply, marked as answer]
I think the easiest way to get around this is to use "@napi-rs/canvas" as mentioned in that issue:
https://github.com/vercel/vercel/issues/3460#issuecomment-994505315

**Reply by xts-bit** [reply]
@swarnava  i tried using @napi-rs/canvas  you mentioned but it gave me An error occurred with your deployment

FUNCTION_INVOCATION_TIMEOUT

**Reply by swarnava** [reply]
I just deployed the app and it seems working as expected: https://nn.vercel-support.app/

**Reply by xts-bit** [reply]
in my case, i am encoding the image in nodejs so its taking a lot of time and vercel give me a timeout I think

**Reply by swarnava** [reply]
You can upgrade the plan for more advanced usecase: https://vercel.com/docs/concepts/limits/overview

**Reply by xts-bit** [reply]
Thank you for your help it works well now the main issue was timeout only
